auth/store.go

```diff
@@ -807,20 +879,53 @@ func (as *authStore) isAuthEnabled() bool {
 	return as.enabled
 }
 
-func NewAuthStore(be backend.Backend) *authStore {
+func NewAuthStore(be backend.Backend, tokenType, jwtSignMethod, jwtPubKeyPath, jwtPrivKeyPath string) *authStore {
```

Comment:
I see. Then how about adding a new struct that includes these parameters? It reduces a conditional branch in etcdserver side.